apps/examples/testcase/le_tc/kernel/kernel_tc_main.c

```diff
@@ -103,7 +103,7 @@ int tc_kernel_main(int argc, char *argv[])
 	libc_misc_main();
 #endif
 
-#ifdef CONFIG_TC_KERNEL_LIBC_MQUEUE
+#if !defined(CONFIG_BUILD_PROTECTED) && defined(CONFIG_TC_KERNEL_LIBC_MQUEUE)
```

Comment:
I don't know why this is needed. The libc_mqueue_main uses syscall APIs or libc APIs. Could you let me know why this is needed?

Reply:
And if this is needed indeed, let's add a dependency in Kconfig, CONFIG_TC_KERNEL_LIBC_MQUEUE. Not here.

Reply:
As libc_mqueue_main has 2 testecases 

1. tc_libc_mqueue_mq_getattr();
2. tc_libc_mqueue_mq_setattr();

and both are giving assert so I directly put main in config check.

Reply:
mq_getattr and setattr are libc APIs. It should not be a problem. Important thing is not a blocking tc, we need to check whether it has a problem or not.

Reply:
Could you check root cause of assert?

Reply:
These testcases internally accessing kernel specific structure to validate which is not allowed in user space in protected build configuration, that's why they are  giving assert. 

So I put this testcase under config check.

Reply:
As suggested I added dependency in Kconfig, Please check